core/base/inc/TNamed.h

```diff
@@ -33,9 +33,9 @@ class TNamed : public TObject {
    TString   fTitle;           //object title
 
 public:
-   TNamed(): fName(), fTitle() { }
+   TNamed() = default;
    TNamed(const char *name, const char *title) : fName(name), fTitle(title) { }
-   TNamed(const TString &name, const TString &title) : fName(name), fTitle(title) { }
+   TNamed(TString name, TString title) : fName(std::move(name)), fTitle(std::move(title)) {}
```

Comment:
Wouldn't
```suggestion
   TNamed(const &TString name, const &TString title) : fName(name), fTitle(title) {}
   TNamed(TString &&name, const TString &title) : fName(std::move(name)), fTitle(title) {}
   TNamed(const TString &name, TString &&title) : fName(name), fTitle(std::move(title)) {}
   TNamed(TString &&name, TString &&title) : fName(std::move(name)), fTitle(std::move(title)) {}
```
be even more performant? (Albeit annoying :) )

Reply:
Please take a look at the discussion in #15434. Especially <https://www.cppstories.com/2018/08/init-string-member/>, which analyzes nearly every combination.

A const-reference is not any more performant then a plain `TString` (assuming move is very cheap).

Reply:
> Especially [cppstories.com/2018/08/init-string-member](https://www.cppstories.com/2018/08/init-string-member/), which analyzes nearly every combination.

Actually they seems have left the case of the rvalue reference/universal forwarding as an exercise to the reader ("For now, I’d like to stop here and don’t go into details. ... " ) ;)

Reply:
> Whenever performance and compiler optimizations are involved it's always best to verify the theoretical claims since it's very hard to guess correctly.

Agreed.  Also note that `TString` does have a "Small String Optimization (SSO)".

Reply:
@silverweed @dpiparo Could check if this change help (or hurts)?